exchanges/binance/binance_websocket.go

```diff
@@ -428,12 +433,24 @@ func (b *Binance) SeedLocalCache(p currency.Pair) error {
 	newOrderBook.Pair = p
 	newOrderBook.AssetType = asset.Spot
 	newOrderBook.ExchangeName = b.Name
+	newOrderBook.LastUpdateID = orderbookNew.LastUpdateID
 
 	return b.Websocket.Orderbook.LoadSnapshot(&newOrderBook)
 }
 
 // UpdateLocalCache updates and returns the most recent iteration of the orderbook
 func (b *Binance) UpdateLocalCache(wsdp *WebsocketDepthStream) error {
+	currencyPair := currency.NewPairFromFormattedPairs(wsdp.Pair, b.GetEnabledPairs(asset.Spot),
+		b.GetPairFormat(asset.Spot, true))
+	currentBook := b.Websocket.Orderbook.GetOrderbook(currencyPair, asset.Spot)
+
+	// Drop any event where u is <= lastUpdateId in the snapshot.
+	// The first processed event should have U <= lastUpdateId+1 AND u >= lastUpdateId+1.
```

Comment:
The second and third parts here are _implied_, but not enforced.

To detect whether the 'first' event actually being processed met this condition would likely involve yet another field on the common orderbook type, which would be specific for Binance handling, hence I've avoided it.

Ensuring that the new event's `U` value is equal to the previous event's `u+1` can be done by comparing to the last update id stored against the orderbook. It's unclear what action to take if this fails however (though the Futures API equivalent would suggest re-seeding and resyncing the WS updates?).

In testing I believe I did see some cases where the `new U != old u+1`, though I've not reproduced that with the current code. Could be a timing/load factor with the servers.